src/extension/cell.ts

```diff
@@ -524,7 +526,9 @@ export class NotebookCellOutputManager {
    *
    */
   async refreshTerminal(terminalState: ITerminalState | undefined): Promise<void> {
-    if (!ContextState.getKey(NOTEBOOK_AUTOSAVE_ON)) {
+    const isSignedIn = features.isOnInContextState(FeatureName.SignedIn)
+
+    if (!isSignedIn) {
```

Comment:
It should be validated only for Stateful, because when you turn on the reporter experiment it's not able to capture the exitCode.